packages/pwa/app/components/_app-config/index.jsx

```diff
@@ -84,6 +72,34 @@ AppConfig.extraGetPropsArgs = (locals = {}) => {
     }
 }
 
+AppConfig.getUserPreferredLocales = (locals) => {
+    // TODO: `locals` as an argument doesn't seem right think a little bit more about this one.
+    // Like do we need the request object here?
+    const {originalUrl} = locals
+
+    const localeUrl = originalUrl && originalUrl.split('/')[1]
+
+    const locale = originalUrl
+        ? SUPPORTED_LOCALES.includes(localeUrl)
+            ? localeUrl
+            : DEFAULT_LOCALE
+        : window?.__PRELOADED_STATE__?.appProps?.targetLocale
+
+    return [locale]
+}
+
+AppConfig.importLocaleData = async (locale) => {
+    let module
+    try {
+        module = await import(`../../translations/compiled/${locale}.json`)
+    } catch (err) {
+        console.error(err)
+        console.log(`Loading the default locale '${DEFAULT_LOCALE}' instead`)
+        module = await import(`../../translations/compiled/${DEFAULT_LOCALE}.json`)
+    }
+    return module.default
+}
```

Comment:
@bendvc I've just added a commit to add this static method to dynamically import the translation file. The import was broken because as you mentioned, the translation files themselves are in the pwa directory.

So I moved some bits of code, in particular this dynamic import logic back into the pwa domain. Now the dynamic import works again.

Reply:
I appreciate that this is a fix. But I'm not sure it's the fix that we want. The goal of this PR is  the abstract the difficult bits of localization into the SDK, doing dynamic imports in the PWA is probably one of the more difficult parts of this design. We'll probably have to think about this a little more and see how we can solve this.

Reply:
We already have another localization-related static method in AppConfig. I think `importLocaleData` is also another localization method that falls under the pwa domain and that references `DEFAULT_LOCALE`. 

And I would say that a dynamic import statement is not complex/difficult to understand. The function is focused on a single thing. On the other hand, `getUserPreferredLocales` is inherently more complex IMO.

Reply:
Another thought: if we move this dynamic loading into the SDK, won't we have to assume the location of the translation files? However, if we have the logic inside the PWA, then developers have control over where the translation files are placed.

Reply:
The idea would be to have translation files live in a configurable folder building upon the idea of having a single app config file. 

Either way, I don't see doing the dynamic loading in the PWA as a viable solution to the problem we are trying to solve, which is moving the implementation into the SDK. Do you get what I trying to say? You suggesting the solution to moving the localization code into the SDK is to move it out back into the PWA.

Reply:
My idea and your idea are actually related/similar, aren't they? :) We both want to extract as much of the localization code out of the PWA and into the SDK. But there are still pieces that have to reside in user land / PWA.

Having the location of this configurable folder saved in the config file, which is located in the PWA, is a piece that needs to be in the PWA. With my idea, the folder location is saved inside the AppConfig's static method.

Reply:
Moving forward, all right, let's move the dynamic loading into the SDK. Do we know whether it's possible to have dynamic loading working from within the SDK? Do we need to investigate it some more?